File: Java/Language/Runtime/ReflectionTest2/Program2.java
class Program {

	public static void main(String[] args) throws Exception {
		double p = Double.parseDouble(args[0]);
		var c = Class.forName(args[1]);
		var interest = c.getMethod(args[2], double.class, int.class);
		var scheme = c.getConstructor().newInstance();
		var md = interest.getAnnotation(finance.MaxDuration.class);
		int m = md != null ? md.value() : 10;
		for(int n = 1; n <= m; ++n){
			float r = (float)interest.invoke(scheme, p, n);
			float i = r / 1200;
			double emi = p * i / (1 - Math.pow(1 + i, -12 * n));
			System.out.printf("%-6d%16.2f%n", n, emi);
		}
	}
}

File: dotnet/Language/Runtime/ReflectionTest2/DemoApp/Program2.cs
using System.Reflection;

delegate float Interest(double amount, int period);

class Program
{
    static void Main(string[] args)
    {
        double p = double.Parse(args[0]);
        Type t = Type.GetType(args[1]);
        MethodInfo scheme = t.GetMethod(args[2]);
        object policy = Activator.CreateInstance(t);
        Interest rate = scheme.CreateDelegate<Interest>(policy);
        int m = 10;
        for(int n = 1; n <= m; ++n)
        {
            float i = rate(p, n) / 1200; 
            double emi = p * i / (1 - Math.Pow(1 + i, -12 * n));
            Console.WriteLine("{0, -6}{1, 12:0.00}", n, emi);
        }
    }
}